library/core/src/str/validations.rs

```diff
@@ -22,7 +22,7 @@ fn utf8_acc_cont_byte(ch: u32, byte: u8) -> u32 {
 /// bits `10`).
 #[inline]
 pub(super) fn utf8_is_cont_byte(byte: u8) -> bool {
-    (byte & !CONT_MASK) == TAG_CONT_U8
+    byte.wrapping_sub(128) < 64
```

Comment:
Even shorter: `(byte as i8) < -64` (although it seems llvm does this transformation by itself).

Reply:
might as well make it more obvious how this needs fewer arithmetic ops